packages/web3-provider/src/subproviders/EthGasStationProvider.js

```diff
@@ -0,0 +1,59 @@
+const fetch = require('cross-fetch')
+const SubProvider = require('web3-provider-engine/subproviders/subprovider')
+
+const CACHE_TIME = 15000
+const GAS_STATION_URL = 'https://ethgasstation.info/json/ethgasAPI.json'
+const GAS_PRICE_KEY = 'safeLow'
+
+class EthGasStationProvider extends SubProvider {
+  constructor() {
+    super()
+
+    this.cache = {
+      data: null,
+      time: null
+    }
+  }
+
+  async fetchData() {
+    const res = await fetch(GAS_STATION_URL)
+    if (res.status !== 200) {
+      throw new Error(`Fetch returned code ${res.status}`)
+    }
+    const jason = await res.json()
+    if (typeof jason[GAS_PRICE_KEY] !== 'undefined') {
+      // values come from EGS as tenths of gwei
+      return String(jason[GAS_PRICE_KEY] * 1e8)
+    }
+    throw new Error(`Gas key of ${GAS_PRICE_KEY} is unavailable`)
+  }
+
+  handleRequest(payload, next, end) {
+    const { method } = payload
+
+    if (method === 'eth_gasPrice') {
+      const now = new Date()
+
+      if (this.cache.time && now - this.cache.time < CACHE_TIME) {
+        return this.cache.data
+      }
+
+      return this.fetchData()
+        .then(data => {
+          this.cache = {
+            data,
+            time: now
+          }
+
+          return end(null, data)
+        })
+        .catch(err => {
+          console.error('Error fetching results from ethgastation.info')
+          console.error(err)
+          return next()
```

Comment:
Do we want to fall back if ethgasstation is down for some reason?

Reply:
`next()` should allow the following subprovider(s) to handle it, essentially skipping this one.  In this case, it'll be ThrottleRPCProvider, which will send it along to the JSON-RPC provider.

Reply:
nit: I had the same question as nick. So maybe add a comment explaining next will cause default web3 gas price implementation to be used. It was not directly obvious to me neither.